external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -252,14 +289,16 @@ private boolean commit() {
     private boolean poll() {
         final int maxUncommittedOffsets = kafkaSpoutConfig.getMaxUncommittedOffsets();
         final int readyMessageCount = retryService.readyMessageCount();
-        final boolean poll = !waitingToEmit() &&
-            //Check that the number of uncommitted, nonretriable tuples is less than the maxUncommittedOffsets limit
-            //Accounting for retriable tuples this way still guarantees that the limit is followed on a per partition basis, and prevents locking up the spout when there are too many retriable tuples
-            (numUncommittedOffsets - readyMessageCount < maxUncommittedOffsets ||
-            consumerAutoCommitMode);
+        final boolean poll = !isWaitingToEmit() &&
+              /*
+               * Check that the number of uncommitted, non-retriable tuples is less than the maxUncommittedOffsets limit
+               * Accounting for retriable tuples this way still guarantees that the limit is followed on a per partition basis,
+               * and prevents locking up the spout when there are too many retriable tuples
+               */
+                (numUncommittedOffsets - readyMessageCount < maxUncommittedOffsets || consumerAutoCommitMode);
```

Comment:
Will do, but this was pre-existing code, not really pertaining to the scope of this patch.